src/desktop/apps/article/components/InfiniteScrollNewsArticle.js

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react'
+import _ from 'underscore'
+import PropTypes from 'prop-types'
+import Waypoint from 'react-waypoint'
+import styled from 'styled-components'
+import { positronql as _positronql } from 'desktop/lib/positronql'
+import { Article } from 'reaction/Components/Publishing'
+import { newsArticlesQuery } from 'desktop/apps/article/queries/articles'
+import { setupFollows, setupFollowButtons } from './FollowButton.js'
+
+// FIXME: Rewire
+let positronql = _positronql
+
+const FETCH_TOP_OFFSET = 200
+
+export default class InfiniteScrollNewsArticle extends React.Component {
+  static propTypes = {
+    article: PropTypes.object,
+    isMobile: PropTypes.bool,
+    marginTop: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isLoading: false,
+      articles: [this.props.article],
+      offset: 0,
+      error: false,
+      following: setupFollows() || null,
+      isEnabled: true,
+    }
+  }
+
+  componentDidMount() {
+    setupFollowButtons(this.state.following)
+  }
+
+  fetchNextArticles = async () => {
+    const { articles, offset } = this.state
+
+    this.setState({
+      isLoading: true,
+    })
+
+    try {
+      const data = await positronql({
+        query: newsArticlesQuery({
+          offset,
+          limit: 3,
```

Comment:
Let's increase this number to at least 10 since news articles are super short. Would probably even feel comfortable going to 20 depending on how the experience feels.